Month1/Week2/Section1/Exercise3/Main3.cpp

```diff
@@ -31,10 +31,17 @@ class FSquare
 {
 private:
 	float SideLenght;
+	
+
 public:
-	FSquare()
+	FSquare() 
+	{
+	}
+	FSquare(float SideLenght, float PerimeterSquare, float AreaSquare)
```

Comment:
Your parameter variables should not be named exactly like your member variables, since now your parameters are overshadowing your members. If you do anything to SideLength here, you will be changing the parameter value, not the member.

Apart from that, I have 3 comments:
1. Why are you asking for parameters if you are going to assign to those parameters the value 0.f and do nothing else. The parameters of constructors should be assigned to your member variables.
2. Why are you asking for a Permiter and an Area? These are not values that the user should input, those are the ones that you calculate inside the class.
3. Your parameters could all be `const float`.

Please take a look at the Constructor video again https://www.youtube.com/watch?v=FXhALMsHwEY&ab_channel=TheCherno

If you have any doubt about what the purpose of a parameter is or what a constructor with parameters should be doing, please let me know! I'll gladly answer them :)